lib/server.js

```diff
@@ -340,6 +352,56 @@ module.exports = function createServer(config, log) {
     }
   )
 
+  api.post('/checkIpOnly', (req, res, next) => {
+    const action = req.body.action
+    const ip = req.body.ip
+
+    if (! action || ! ip) {
+      const err = { code: 'MissingParameters', message: 'action and ip are both required' }
+      log.error({ op:'request.checkAuthenticated', action: action, ip: ip, err: err })
+      res.send(400, err)
+      return next()
+    }
+
+    fetchRecords(ip)
+      .spread((ipRecord, reputation) => {
+        if (ipRecord.isBlocked()) {
+          return {
+            block: true,
+            retryAfter: ipRecord.retryAfter()
+          }
+        }
+
+        let retryAfter = ipRecord.update(action)
+        let block = retryAfter > 0
+        let suspect = requestChecks.treatEveryoneWithSuspicion || reputationService.isSuspectBelow(reputation)
+        let blockReason
+
+        if (block) {
+          blockReason = blockReasons.OTHER
+        }
+
+        if (config.ipBlocklist.enable && blockListManager.contains(ip)) {
+          block = true
+          blockReason = blockReasons.IP_IN_BLOCKLIST
+          retryAfter = 0
+        }
+
+        if (reputationService.isBlockBelow(reputation)) {
+          block = true
+          retryAfter = ipRecord.retryAfter()
+          blockReason = blockReasons.IP_BAD_REPUTATION
+        }
+
+        return setRecords(ip, ipRecord)
+          .then(() => ({ block, blockReason, retryAfter, suspect }))
+      })
+      .then(result => {
+        res.send(result)
```

Comment:
I think we should add a check for whitelisted IPs, similar to https://github.com/mozilla/fxa-customs-server/blob/96858c0c028e10bf521fe684120d40d88d1b0553/lib/server.js#L247. Maybe we pull this into some common place for use.